framework/src/meshgenerators/BreakMeshByBlockGenerator.C

```diff
@@ -50,22 +51,44 @@ BreakMeshByBlockGenerator::validParams()
 BreakMeshByBlockGenerator::BreakMeshByBlockGenerator(const InputParameters & parameters)
   : BreakMeshByBlockGeneratorBase(parameters),
     _input(getMesh("input")),
-    _block(parameters.isParamSetByUser("block") ? getParam<std::vector<SubdomainID>>("block")
-                                                : std::vector<SubdomainID>()),
-    _block_set(_block.begin(), _block.end()),
-    _block_restricted(parameters.isParamSetByUser("block")),
+    _surrounding_blocks(parameters.isParamSetByUser("surrounding_blocks")
+                            ? getParam<std::vector<SubdomainID>>("surrounding_blocks")
+                            : std::vector<SubdomainID>()),
+    _block_pairs(parameters.isParamSetByUser("block_pairs")
+                     ? getParam<std::vector<std::vector<SubdomainID>>>("block_pairs")
+                     : std::vector<std::vector<SubdomainID>>()),
+    _block_pairs_restricted(parameters.isParamSetByUser("block_pairs")),
+    _surrounding_blocks_restricted(parameters.isParamSetByUser("surrounding_blocks")),
```

Comment:
shouldn't we first initialize the `_surrounding_blocks_restricted` and `_block_pairs_restricted` and then use them to get the right paramters?

Reply:
looks like `set` and `pair` are not supported. Let us use `vector` for now.

For our case, the code should work fine even if we have duplicated items.

Reply:
Ok, just use a vector for the input and store them in std::set<std::pair>, so duplicates will be removed automatically

Reply:
That works. Do you want to throw an error or just remove the duplication. Like I explained, duplication does not matter. If you think an error is necessary here, I am happy to make that change.